lib/src/main/kotlin/app/cash/kfsm/MachineBuilder.kt

```diff
@@ -74,6 +75,29 @@ class MachineBuilder<ID, V : Value<ID, V, S>, S : State<ID, V, S>> {
     internal fun build(): Map<S, Transition<ID, V, S>> = transitions
   }
 
+  /**
+   * Defines transitions from a given state using a builder block.
+   *
+   * @param selector Given a value, select the next appropriate states
+   * @param block A builder block that defines the possible transitions from this state
+   * @throws IllegalStateException if transitions for this state have already been defined
```

Comment:
Out of curiosity, why did we opt to throwing exceptions, rather than returning a (Kotlin native) `Result`? Not like we're forcing anyone to use Arrow by doing so.

Reply:
This is within the builder DSL, which is mutating maps as it runs. The Result type is returned on build()